lib/src/map_mutations.dart

```diff
@@ -137,6 +137,8 @@ SourceEdit _replaceInBlockMap(
       !isFlowYamlCollectionNode(newValue) &&
       !isEmpty(newValue)) {
     valueAsString = lineEnding + valueAsString;
+  } else {
+    valueAsString = ' ' + valueAsString;
   }
```

Comment:
Wouldn't it be safer / slightly more correct to do:
```dart
if (!valueAsString.startsWith(lineEnding)) {
  // prepend whitespace to ensure there is space after colon.
  valueAsString = ' ' + valueAsString;
}
```

Reply:
I'm mostly thinking what if one day `yamlEncodeBlockString` returns something that starts with a newline?

Reply:
Yes. This would be more safer.